evalml/tests/automl_tests/test_automl_iterative_algorithm.py

```diff
@@ -0,0 +1,1146 @@
+
+
+@pytest.mark.parametrize("return_dict", [True, False])
+def test_describe_pipeline_with_ensembling_iterative(
+    return_dict, X_y_binary, AutoMLTestEnv, caplog
+):
+    X, y = X_y_binary
+    two_stacking_batches = 1 + 2 * (len(get_estimators(ProblemTypes.BINARY)) + 1)
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        max_batches=two_stacking_batches,
+        objective="Log Loss Binary",
+        ensembling=True,
+        optimize_thresholds=False,
+        error_callback=raise_error_callback,
+        _automl_algorithm="iterative",
+    )
+
+    score_side_effect = [
+        {"Log Loss Binary": score}
+        for score in np.arange(
+            0, -1 * automl.max_iterations * automl.data_splitter.get_n_splits(), -0.1
+        )
+    ]  # Dcreases with each call
+
+    test_env = AutoMLTestEnv("binary")
+    with test_env.test_context(mock_score_side_effect=score_side_effect):
+        automl.search()
+    pipeline_names = automl.rankings["pipeline_name"]
+    assert pipeline_names.str.contains("Ensemble").any()
+
+    ensemble_ids = [
+        _get_first_stacked_classifier_no() - 1,
+        len(automl.results["pipeline_results"]) - 1,
+    ]
+
+    for i, ensemble_id in enumerate(ensemble_ids):
+        caplog.clear()
+        automl_dict = automl.describe_pipeline(ensemble_id, return_dict=return_dict)
+        out = caplog.text
+        assert "Stacked Ensemble Classification Pipeline" in out
+        assert "* final_estimator : Elastic Net Classifier" in out
+        assert "Problem Type: binary" in out
+        assert "Model Family: Ensemble" in out
+        assert "Total training time (including CV): " in out
+        assert "Log Loss Binary # Training # Validation" in out
+        assert "Input for ensembler are pipelines with IDs:" in out
+
+        if return_dict:
+            assert automl_dict["id"] == ensemble_id
+            assert (
+                automl_dict["pipeline_name"]
+                == "Stacked Ensemble Classification Pipeline"
+            )
+            assert "Stacked Ensemble Classifier" in automl_dict["pipeline_summary"]
+            assert isinstance(automl_dict["mean_cv_score"], float)
+            assert not automl_dict["high_variance_cv"]
+            assert isinstance(automl_dict["training_time"], float)
+            assert isinstance(
+                automl_dict["percent_better_than_baseline_all_objectives"], dict
+            )
+            assert isinstance(automl_dict["percent_better_than_baseline"], float)
+            assert isinstance(automl_dict["validation_score"], float)
+            assert len(automl_dict["input_pipeline_ids"]) == len(
+                allowed_model_families("binary")
+            )
+            if i < 2:
+                assert all(
+                    input_id < ensemble_id
+                    for input_id in automl_dict["input_pipeline_ids"]
+                )
+            else:
+                assert all(
```

Comment:
There's some coverage issues in this file that I can't figure out as I didn't modify these tests other than expliceily calling `_automl_algorithm='iterative'`. I believe that there was coverage drift but we never noticed as the coverage changes were too small. Could be wrong here so would appreciate some help here.

Reply:
Checking code cov its apparent that this coverage is[ missing on main as well ](https://app.codecov.io/gh/alteryx/evalml/blob/main/evalml/tests/automl_tests/test_automl.py). One can look at the link and search up the test without the `_iterative` tag such as `test_describe_pipeline_with_ensembling` here.